worlds/pokemon_emerald/data.py

```diff
@@ -409,407 +492,408 @@ def _init() -> None:
 
     # Excludes extras like copies of Unown and special species values like SPECIES_EGG.
     all_species: List[Tuple[str, str]] = [
```

Comment:
For 1, good catch, changed in [feae432](https://github.com/ArchipelagoMW/Archipelago/pull/2918/commits/feae432874886e897b5306c3722b1f67589a4ace).

For 2 and 3, its sort of the same answer to both. I believe the Gen 3 pokemon data was built off or copypasted from Gen 2 at the beginning of development. So The first 251 entries are in the same order they were in Gen 2, and that order was retained for the national pokedex moving forward.

Appended to the Gen 2 species list are 25 alternate forms of Unown. I don't really know why, or if they're even referenced in Emerald, but they were left in, and occupy internal ids 252 through 276. Treecko is the first Gen 3 pokemon, with internal id 277, and then ids increment and are contiguous in _Hoenn_ pokedex order, which is distinct from national pokedex order. And then again, 27 alternate forms of Unown are appended to the end for use in Gen 3, now including the ? and ! variants.

All that to say, the internal id for any given species is not necessarily related to its national pokedex id. National id is more or less an extra property that happens to be aligned with internal id in a lot of cases. But, it's now relevant for the apworld to know in order to communicate with other generations, since national id is always consistent.

Reply:
Well, that's partly the point. This is explicitly every species that the randomizer should care about, which means ignoring the duplicate Unown data. The internal id (once we index into `data.constants` with the first item), user-friendly name, and national id. None of the three columns can be derived from the other, and the internal ids, which I use as the true identifier, are not contiguous.